pkg/resources/aws/aws.go

```diff
@@ -530,16 +530,34 @@ func DumpInstance(op *resources.DumpOperation, r *resources.Resource) error {
 	data["raw"] = r.Obj
 	op.Dump.Resources = append(op.Dump.Resources, data)
 
+	i := GetInstanceFromResource(r)
+	ec2Instance := r.Obj.(*ec2.Instance)
+	imageID := aws.StringValue(ec2Instance.ImageId)
+	imageInfo, err := getDumpState(op).getImageInfo(imageID)
+	if err != nil {
+		klog.Warningf("unable to fetch image %q: %v", imageID, err)
+	} else if imageInfo != nil {
+		i.SSHUser = imageInfo.SSHUser
+	}
+	op.Dump.Instances = append(op.Dump.Instances, i)
+
+	return nil
+}
+
+func GetInstanceFromResource(r *resources.Resource) *resources.Instance {
 	ec2Instance := r.Obj.(*ec2.Instance)
 	i := &resources.Instance{
-		Name: r.ID,
+		ID:   r.ID,
+		Name: r.Name,
 	}
 	for _, networkInterface := range ec2Instance.NetworkInterfaces {
 		if networkInterface.Association != nil {
 			publicIP := aws.StringValue(networkInterface.Association.PublicIp)
 			if publicIP != "" {
 				i.PublicAddresses = append(i.PublicAddresses, publicIP)
 			}
+			i.PrivateAddress = aws.StringValue(networkInterface.PrivateIpAddress)
+
```

Comment:
Private IP is not related to Association.
```suggestion
		i.PrivateAddress = aws.StringValue(networkInterface.PrivateIpAddress)
		if networkInterface.Association != nil {
			publicIP := aws.StringValue(networkInterface.Association.PublicIp)
			if publicIP != "" {
				i.PublicAddresses = append(i.PublicAddresses, publicIP)
			}
```